pkg/build/builder/cmd/builder.go

```diff
@@ -106,19 +110,21 @@ func setupSourceSecret(sourceSecretName string, scmAuths []scmauth.SCMAuth) erro
 	}
 
 	if len(scmAuthsPresent) == 0 {
-		return fmt.Errorf("no auth handler was found for the provided secret %q",
+		return nil, fmt.Errorf("no auth handler was found for the provided secret %q",
 			sourceSecretName)
 	}
 
+	url := &url.URL{}
+
 	for name, auth := range scmAuthsPresent {
 		glog.V(3).Infof("Setting up SCMAuth %q", name)
-		if err := auth.Setup(sourceSecretDir); err != nil {
+		if url, err = auth.Setup(sourceSecretDir); err != nil {
```

Comment:
actually I would use a temp variable here and only if it's non-nil assign it to the outer variable. Otherwise if you have multiple scmAuth's, one that returns a nil value will overwrite the previous scmAuth's URL. I would also either log a warning or throw an error if multiple scmAuth's return non-nil for the URL.